### PR TITLE
Tiny compiler fixes

### PR DESCRIPTION
**Pull request by roman-khimov**
Just enough to make 0.104.0 more fun.

### CODE REVIEW

File: pkg/compiler/import_test.go [hunk 1]
```diff
@@ -61,3 +65,27 @@ func TestImportNameSameAsOwn(t *testing.T) {
 	}`
 	eval(t, src, big.NewInt(3))
 }
+
+func TestImportCycleIndirect(t *testing.T) {
```
Comment (AnnaShaleva) [inline pkg/compiler/import_test.go:69]: This one should be probably called `TestImportCycleDirect` (pkg2 imports pkg3 and pkg 3 imports pkg2).

File: pkg/compiler/import_test.go [hunk 2]
```diff
@@ -61,3 +65,27 @@ func TestImportNameSameAsOwn(t *testing.T) {
...
+		import "github.com/nspcc-dev/neo-go/pkg/compiler/testdata/importcycle/pkg2"
+		func Main() int {
+			return pkg2.A
+		}
+	`
+	_, _, err := compiler.CompileWithOptions("some.go", strings.NewReader(src), nil)
+	require.Error(t, err)
+}
+
+func TestImportCycleDirect(t *testing.T) {
```
Comment (AnnaShaleva) [inline pkg/compiler/import_test.go:81]: And this one is `Indirect` (pkg1 imports pkg2, and import cycle is in pkg2-pkg3).

But it's a nitpicking, tests are good.

File: pkg/compiler/compiler.go [hunk 3]
```diff
@@ -357,8 +358,16 @@ func CompileAndSave(src string, o *Options) ([]byte, error) {
...
+				var keys = make([]string, 0, len(di.EmittedEvents))
+				for k := range di.EmittedEvents {
+					keys = append(keys, k)
+				}
+				sort.Strings(keys)
+				for _, eventName := range keys {
+					var (
+						eventUsages   = di.EmittedEvents[eventName]
+						manifestEvent HybridEvent
+					)
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/compiler/compiler.go:370]: They're a part of proper `NamedTypes`, their names would be X-ified in `scAndVMTypeFromType`. But they won't be stable, true. Fixed.

File: pkg/compiler/import_test.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/compiler/import_test.go]: Ah, yeah, fixed.

**Review by AnnaShaleva — APPROVED** [review]
Makes life better.

**Pull request merged**